File: create_tables/seis_activity.py
import oracledb
import sys
import lib_platform as platform

if platform.system == "darwin":
    sys.path.insert(0, "database")
elif platform.system == "windows":
    sys.path.insert(0, "../database")

from conn import connection

# pw = getpass.getpass("Enter password: ")

# connection = oracledb.connect(
#     user="admin",
#     password="ppdm-3.9",
#     dsn="/kei-oracle-test.chkxr21l2l5s.ap-southeast-3.rds.amazonaws.com:1521/ORCL")

# print("Successfully connected to Oracle Database")

cursor = connection.cursor()


cursor.execute("""
    begin
        execute immediate 'drop table seis_activity';
        exception when others then if sqlcode <> -942 then raise; end if;
    end;""")

cursor.execute(
    """create table seis_activity(
        SEIS_SET_SUBTYPE VARCHAR(30) NOT NULL,
        SEIS_SET_ID VARCHAR(40) NOT NULL,
        ACTIVITY_OBS_NO INT,
        ACTIVE_IND VARCHAR(1),
        ACTIVITY_DURATION DECIMAL(10,5),
        ACTIVITY_DURATION_OUOM VARCHAR(40),
        AREA_SIZE DECIMAL(20,10),
        AREA_SIZE_OUOM VARCHAR(40),
        CREW_COMPANY_BA_ID VARCHAR(40),
        CREW_ID VARCHAR(40),
        EFFECTIVE_DATE TIMESTAMP(0),
        END_DATE TIMESTAMP(0),
        END_TIME TIMESTAMP(0),
        END_TIMEZONE VARCHAR(40),
        EXPIRY_DATE TIMESTAMP(0),
        FIRST_NLINE_NO INT,
        FIRST_SEIS_POINT_ID VARCHAR(16),
        FIRST_XLINE_NO INT,
        LAST_NLINE_NO INT,
        LAST_SEIS_POINT_ID VARCHAR(16),
        LAST_XLINE_NO INT,
        LINE_LENGTH DECIMAL(10,3),
        LINE_LENGTH_OUOM VARCHAR(40),
        OWNER_BA_ID VARCHAR(40),
        PPDM_GUID VARCHAR(38),
        REMARK VARCHAR(2000),
        SEIS_ACTIVITY_CLASS VARCHAR(40),
        SEIS_ACTIVITY_TYPE VARCHAR(40),
        SOURCE VARCHAR(40),
        START_DATE TIMESTAMP(0),
        START_TIME TIMESTAMP(0),
        START_TIMEZONE VARCHAR(40),
        TOTAL_TIME_ELAPSED DECIMAL(10,5),
        TOTAL_TIME_ELAPSED_OUOM VARCHAR(40),
        ROW_CHANGED_BY VARCHAR(30),
        ROW_CHANGED_DATE TIMESTAMP(0),
        ROW_CREATED_BY VARCHAR(30),
        ROW_CREATED_DATE TIMESTAMP(0),
        ROW_EFFECTIVE_DATE TIMESTAMP(0),
        ROW_EXPIRY_DATE TIMESTAMP(0),
        ROW_QUALITY VARCHAR(40))""")

File: postgres_39/all_struct_with_default/SeisActivity.go
package dto

type Seis_activity struct {
	Seis_set_subtype        string   `json:"seis_set_subtype" default:"source"`
	Seis_set_id             string   `json:"seis_set_id" default:"source"`
	Activity_obs_no         int      `json:"activity_obs_no" default:"1"`
	Active_ind              *string  `json:"active_ind" default:""`
	Activity_duration       *float64 `json:"activity_duration" default:""`
	Activity_duration_ouom  *string  `json:"activity_duration_ouom" default:""`
	Area_size               *float64 `json:"area_size" default:""`
	Area_size_ouom          *string  `json:"area_size_ouom" default:""`
	Crew_company_ba_id      *string  `json:"crew_company_ba_id" default:""`
	Crew_id                 *string  `json:"crew_id" default:""`
	Effective_date          *string  `json:"effective_date" default:""`
	End_date                *string  `json:"end_date" default:""`
	End_time                *string  `json:"end_time" default:""`
	End_timezone            *string  `json:"end_timezone" default:""`
	Expiry_date             *string  `json:"expiry_date" default:""`
	First_nline_no          *int     `json:"first_nline_no" default:""`
	First_seis_point_id     *string  `json:"first_seis_point_id" default:""`
	First_xline_no          *int     `json:"first_xline_no" default:""`
	Last_nline_no           *int     `json:"last_nline_no" default:""`
	Last_seis_point_id      *string  `json:"last_seis_point_id" default:""`
	Last_xline_no           *int     `json:"last_xline_no" default:""`
	Line_length             *float64 `json:"line_length" default:""`
	Line_length_ouom        *string  `json:"line_length_ouom" default:""`
	Owner_ba_id             *string  `json:"owner_ba_id" default:""`
	Ppdm_guid               *string  `json:"ppdm_guid" default:""`
	Remark                  *string  `json:"remark" default:""`
	Seis_activity_class     *string  `json:"seis_activity_class" default:""`
	Seis_activity_type      *string  `json:"seis_activity_type" default:""`
	Source                  *string  `json:"source" default:""`
	Start_date              *string  `json:"start_date" default:""`
	Start_time              *string  `json:"start_time" default:""`
	Start_timezone          *string  `json:"start_timezone" default:""`
	Total_time_elapsed      *float64 `json:"total_time_elapsed" default:""`
	Total_time_elapsed_ouom *string  `json:"total_time_elapsed_ouom" default:""`
	Row_changed_by          *string  `json:"row_changed_by" default:""`
	Row_changed_date        *string  `json:"row_changed_date" default:""`
	Row_created_by          *string  `json:"row_created_by" default:""`
	Row_created_date        *string  `json:"row_created_date" default:""`
	Row_effective_date      *string  `json:"row_effective_date" default:""`
	Row_expiry_date         *string  `json:"row_expiry_date" default:""`
	Row_quality             *string  `json:"row_quality" default:""`
}
